rtl/eclrtl/rtlnewkey.cpp

```diff
@@ -2249,39 +2249,22 @@ void RowFilter::splitIntoKeyFilter(const RtlRecord & record, RowFilter &keyFilte
 {
     if (0 == filters.ordinality())
         return;
-    IConstArrayOf<IFieldFilter> temp;
-    ForEachItemIn(i, filters)
-        temp.append(OLINK(filters.item(i)));
-    temp.sort(compareFieldFilters);
 
     unsigned numKeyedFields = record.getNumKeyedFields();
-    unsigned fieldNum = 0;
-    unsigned filterIdx = 0;
-    while (true)
-    {
-        const IFieldFilter & cur = temp.item(filterIdx);
-        unsigned curFilterField = cur.queryFieldIndex();
-        if (curFilterField >= numKeyedFields)
-            break;
-        while (fieldNum < curFilterField)
-        {
-            keyFilter.addFilter(*createWildFieldFilter(fieldNum, *record.queryType(fieldNum)));
-            fieldNum++;
-        }
-        keyFilter.addFilter(OLINK(cur));
-        if (++filterIdx == temp.ordinality())
-            break;
-        fieldNum++;
-    }
-    if (keyFilter.numFilterFields()) // if none, then all can remain in original filter
+    IConstArrayOf<IFieldFilter> keyFilters;
+    ForEachItemInRev(i, filters)
     {
-        clear();
-        while (filterIdx < temp.ordinality())
+        const IFieldFilter & cur = filters.item(i);
+        if (cur.queryFieldIndex() < numKeyedFields)
         {
-            addFilter(OLINK(temp.item(filterIdx)));
-            filterIdx++;
+            keyFilters.append(OLINK(cur));
+            filters.remove(i);
         }
     }
+    keyFilter.sortByFieldOrder(); // NB: need to be ordered ahead of keyFilter.createSegmentMonitors being used
```

Comment:
line 2260 has added to keyFilters not keyFilter - so the keyed filters will be thrown away